mailpoet/lib/Automation/Integrations/WooCommerce/Subjects/ProductsSubject.php

```diff
@@ -0,0 +1,59 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Automation\Integrations\WooCommerce\Subjects;
+
+use MailPoet\Automation\Engine\Data\Subject as SubjectData;
+use MailPoet\Automation\Engine\Exceptions\InvalidStateException;
+use MailPoet\Automation\Engine\Integration\Payload;
+use MailPoet\Automation\Engine\Integration\Subject;
+use MailPoet\Automation\Integrations\WooCommerce\Payloads\ProductsPayload;
+use MailPoet\Validator\Builder;
+use MailPoet\Validator\Schema\ObjectSchema;
+use MailPoet\WooCommerce\Helper as WooCommerceHelper;
+
+/**
+ * @implements Subject<ProductsPayload>
+ */
+class ProductsSubject implements Subject {
```

Comment:
Do you think the `product_ids` can be in the abandoned cart subject? Is there any reason to have this in two different subjects?

Reply:
consolidating discussion here: https://github.com/mailpoet/mailpoet-premium/pull/708#discussion_r1165032856